### PR TITLE
Vapor 4 Migration

### PR DESCRIPTION
**Pull request by ptoffy**
This aims to provide support for vapor 4

### CODE REVIEW

**Comment by ptoffy** [comment]
cd7bc902b33d62cf7392d8466f64429aa399e9b9 looks like a big commit but it's really just a reorganisation of files and structs, plus I re-added some logic to send an email to a newly created user if they need to create their own password

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: CI
+
+on:
+  push:
+    branches: ["master"]
```
Comment (0xTim) [inline .github/workflows/ci.yml]:
```suggestion
    branches: ["master", "main"]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  pull_request:
+    branches: ["master"]
```
Comment (0xTim) [inline .github/workflows/ci.yml]: We'll do a branch rename soon

```suggestion
    branches: ["master", "main"]
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  build:
+    container:
+      image: swift:5.7
```
Comment (0xTim) [inline .github/workflows/ci.yml]:
```suggestion
      image: swift:5.8
```

File: Sources/AdminPanel/Controllers/DashboardController.swift [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
-    public func renderDashboard(_ req: Request) throws -> Future<Response> {
-        let config = try req.make(AdminPanelConfig<U>.self)
-
-        return try req
-            .view()
-            .render(config.views.dashboard.index, on: req)
-            .encode(for: req)
+public final class DashboardController: DashboardControllerType {
+    public func boot(routes: RoutesBuilder) throws {
+        routes.get("", use: dashboardHandler)
```
Comment (0xTim) [inline Sources/AdminPanel/Controllers/DashboardController.swift]:
```suggestion
        routes.get(use: dashboardHandler)
```

File: Sources/AdminPanel/Controllers/AdminPanelUserController.swift [hunk 5]
```diff
@@ -1,172 +1,166 @@
...
+        
+        let resetToken = try user.generateResetPasswordToken(context: .newUserWithoutPassword)
+        try await resetToken.save(on: req.db)
+        
+        let url = req.adminPanel.config.baseURL.appending(
+            "\(req.adminPanel.config.endpoints.resetPassword)/\(resetToken)"
+        )
+                
+        if userCreate.shouldSpecifyPassword == false {
+            _ = try await req.adminPanel.requestPasswordReset(
```
Comment (0xTim) [inline Sources/AdminPanel/Controllers/AdminPanelUserController.swift]: We should mark this with `@discardableResult` if we most likely won't use it in most places, or change the function to return nothing (and throw on error)

**Review by 0xTim — APPROVED** [review]
Few comments but looks good overall!

**Pull request merged**